plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -453,6 +449,20 @@ public Optional<Void> rewriteExpression(ConnectorExpression expression)
         return rewriteResult;
     }
 
+    @Override
+    public Optional<TableFunctionApplicationResult<ConnectorTableHandle>> applyTableFunction(ConnectorSession session, ConnectorTableFunctionHandle handle)
+    {
+        if (!(handle instanceof QueryFunctionHandle)) {
+            return Optional.empty();
+        }
+
+        PinotTableHandle tableHandle = (PinotTableHandle) ((QueryFunctionHandle) handle).getTableHandle();
+        checkState(tableHandle.getQuery().isPresent(), "dynamic table is not present");
```

Comment:
It would be better to avoid "dynamic table" in query pass-through table function.

Reply:
Agreed. The end result should be that we can deprecate and eventually get rid of dynamic table functionality since it's complex to maintain (and easy to introduce different behaviour/results between queries with or without dynamic tables).

Reply:
We will eventually not expose dynamic table to users, i.e. no more "pass through" queries as identifiers. Still need the class as it determines the types, allows for pushdown etc. lmk if you have some ideas.

Reply:
I personally would probably copy over the minimum needed functionality into a new class. That way any changes for PTF won't affect dynamic tables and it'll be easier to deprecate and eventually remove the dynamic table class.

Reply:
Sure! That will simplify the code a lot. Will update.

Reply:
re: the different result: that is because the first query is a broker query with a limit of 10 and the second query is a server query which has a limit of 12 - @ebyhr